modules/apps/journal/journal-web/src/test/java/com/liferay/journal/web/internal/dao/search/JournalRecentArticlesResultRowSplitterTest.java

```diff
@@ -0,0 +1,241 @@
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.TimeZone;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.mockito.Mockito;
+
+/**
+ * @author Jürgen Kappler
+ */
+public class JournalRecentArticlesResultRowSplitterTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Before
+	public void setUp() {
+		LanguageUtil languageUtil = new LanguageUtil();
+
+		Mockito.when(
+			_language.get(Mockito.any(Locale.class), Mockito.anyString())
+		).thenAnswer(
+			invocation -> invocation.getArguments()[1]
+		);
+
+		languageUtil.setLanguage(_language);
+	}
+
+	@Test
+	public void testSplitMultipleSections() {
+		ThemeDisplay themeDisplay = _getThemeDisplay();
+
+		LocalDateTime localDateTime = _toLocalDateTime(
+			new Date(), ZoneId.of("UTC"));
+
+		JournalArticle todayJournalArticle = _getJournalArticle(
+			0, localDateTime);
+		JournalArticle oneDayAgoJournalArticle = _getJournalArticle(
+			1, localDateTime);
+		JournalArticle sevenDaysAgoJournalArticle = _getJournalArticle(
+			7, localDateTime);
+		JournalArticle eightDaysAgoJournalArticle = _getJournalArticle(
+			8, localDateTime);
+		JournalArticle thirtyDaysAgoJournalArticle = _getJournalArticle(
+			30, localDateTime);
+		JournalArticle thirtyOneDaysAgoJournalArticle = _getJournalArticle(
+			31, localDateTime);
+
+		List<ResultRow> resultRows = Arrays.asList(
+			_getResultRow(todayJournalArticle),
+			_getResultRow(oneDayAgoJournalArticle),
+			_getResultRow(sevenDaysAgoJournalArticle),
+			_getResultRow(eightDaysAgoJournalArticle),
+			_getResultRow(thirtyDaysAgoJournalArticle),
+			_getResultRow(thirtyOneDaysAgoJournalArticle));
+
+		JournalRecentArticlesResultRowSplitter
+			journalRecentArticlesResultRowSplitter =
+				new JournalRecentArticlesResultRowSplitter(themeDisplay);
+
+		List<ResultRowSplitterEntry> resultRowSplitterEntries =
+			journalRecentArticlesResultRowSplitter.split(resultRows);
+
+		Assert.assertEquals(
+			resultRowSplitterEntries.toString(), 4,
+			resultRowSplitterEntries.size());
+
+		_assertResultRowSplitterEntry(
+			"today", resultRowSplitterEntries.get(0), themeDisplay,
+			todayJournalArticle);
+		_assertResultRowSplitterEntry(
+			"last-7-days", resultRowSplitterEntries.get(1), themeDisplay,
+			oneDayAgoJournalArticle, sevenDaysAgoJournalArticle);
+		_assertResultRowSplitterEntry(
+			"last-30-days", resultRowSplitterEntries.get(2), themeDisplay,
+			eightDaysAgoJournalArticle, thirtyDaysAgoJournalArticle);
+		_assertResultRowSplitterEntry(
+			"older", resultRowSplitterEntries.get(3), themeDisplay,
+			thirtyOneDaysAgoJournalArticle);
+	}
+
+	@Test
+	public void testSplitSingleSection() {
+		ThemeDisplay themeDisplay = _getThemeDisplay();
+
+		LocalDateTime localDateTime = _toLocalDateTime(
```

Comment:
Only this small change, maybe we could add inline like this :

Reply:
JournalArticle thirtyOneDaysAgoJournalArticle = _getJournalArticle(
			31, _toLocalDateTime(
				new Date(), ZoneId.of("UTC")));

Reply:
Done @Mikellorza, thanks!